src/java/org/apache/cassandra/gms/EndpointState.java

```diff
@@ -110,10 +116,25 @@ public void update(Set<Map.Entry<ApplicationState, VersionedValue>> entries)
         if (updater == null)
             return;
 
-        updater.accept(info -> {
-            entries.forEach(e -> updateNodeInfo(info, e.getKey(), e.getValue()));
-            return info;
-        });
+        ThrowingFunction<NodeInfo<?>, NodeInfo<?>, UnknownHostException> identity = v -> v;
+        List<ThrowingFunction<NodeInfo<?>, NodeInfo<?>, UnknownHostException>> allUpdates = entries.stream()
+                                                                                                     .map(e -> ObjectUtils.firstNonNull(updateNodeInfo(e.getKey(), e.getValue()), identity)).collect(Collectors.toList());
```

Comment:
this looks soo complex...
what do you think about something more like:

```
allUpdates = entries.stream()
    .map(e -> updateNodeInfo(e.getKey(), e.getValue())
    .filter(e -> e != null) // or identity if you change null to identity in updateNodeInfo
    .collect(Collectors.toList());

if (!allUpdates.isEmpty())
{
   // don't even reduce, just use forEach, like:
   updater.accept(info -> {
      // I didn't put exception handling etc.
        allUpdates.forEach(update -> update.apply(info));
   })
}
```

I also don't like these functional interfaces much. I mean - they come from a rather obscure, unrelated library... I'm not sure what to do about that...
One more problem with these interfaces is that the crucial part of `updateNodeInfo` is that it modifies the state in place, whereas the interface may suggest that it returns a copy (i.e. that node info is immutable).